### ISSUE TITLE
Trivy Scans persisted to Defect Dojo are missing multiple metadata fields

### ISSUE THREAD

**Issue by nixboot** [secureCodeBox/secureCodeBox#2364]
### Describe the bug

When SecureCodeBox uploads a Trivy scan to Defect Dojo it is uploaded as the type `Generic Findings Import` and not `Trivy Scan`.

This seems to have the effect that most metadata about the finding are not set. 
E.g. CWE, Vulnerability ID (CVE), Location etc.

In some cases (I have seen it 1 time) severity is even reported wrongly (as High instead of Critical). See the screenshot below.

This is both an issue with `trivy image` and `trivy k8s`. Probably the other types as well.

### Steps To Reproduce

1. Install the [operator](https://artifacthub.io/packages/helm/securecodebox/operator), the [Trivy scanner](https://artifacthub.io/packages/helm/securecodebox/trivy) and the [persistence dojo hook](https://artifacthub.io/packages/helm/securecodebox/persistence-defectdojo) in the below mentioned versions.

2. Start a scan by applying the following scan

```yaml
apiVersion: "execution.securecodebox.io/v1"
kind: Scan
metadata:
  name: "juice-shop"
spec:
  scanType: "trivy-image"
  parameters:
    - bkimminich/juice-shop:v10.2.0
```

3. Open a finding in Defect Dojo.

4. See that CWE is not set, the no tags are set, the location is not set etc. 

5. (See the attached screenshot for a comparison between a manually uploaded Trivy scan and the SCB uploaded Trivy scan)

### Expected behavior
The scan is being uploaded with the type Trivy scan, and metadata findings are populated as if I uploaded the scan manually.

### System (please complete the following information):
- operator: 4.4.1
- persistence-defectdojo: 1.12.0
- Trivy 0.49.1
- DefectDojo version: 2.31.5
- Kubernetes version 1.28.3

For the Operator, Trivy and persistence-defectdojo I am using the Helm chart version 4.4.1 with default values.

### Screenshots 
The screenshot compares a SCB uploaded Trivy finding with a manually uploaded Trviy finding.
It is the same finding, in the same deployment, scanned using the same version of Trivy in the same version. 

![SCR-20240314-mgpx](https://github.com/secureCodeBox/secureCodeBox/assets/112619895/3dddc6be-e9c1-4972-9e87-e16d325d0679)

### Additional context

I tried to upload the Trivy scan that was saved in MinIO with import type Trivy scan. This worked as expected: metadata fields are populated and the severity is correctly set.

**Comment by nixboot** [comment]
If we can find the cause of the issue, we would be willing to contribute a fix for it.

**Comment by Weltraumschaf** [comment]
Hi @nixboot,

thanks for submitting this issue. We'll investigate this.

**Comment by J12934** [comment]
Hi @nixboot 

I'm relativly sure that this is caused by a missing mapping in the mapping between SCB ScanType and DefectDojo tool names which is found here: https://github.com/secureCodeBox/secureCodeBox/blob/main/hooks/persistence-defectdojo/hook/src/main/java/io/securecodebox/persistence/util/ScanNameMapping.java#L17

It still only has the single "trivy" scantype in it, which has been split into the `trivy-image`, `trivy-filesystem` and so on scantypes. Basically any of the names which can be found in this file: https://github.com/secureCodeBox/secureCodeBox/blob/main/scanners/trivy/templates/trivy-scan-type.yaml#L8

Do you know if DefectDojo can properly handle the trivy k8s output? The format is quit a bit different than the normal trivy output if i remember correctly, we might want to still do our generic mapping if DefectDojo doesn't properly support it natively.

**Comment by nixboot** [comment]
Hi @J12934 

Yes, I believe it can handle `trivy k8s` output - at least in the test I've done :) 

Manually uploading the results of 

```bash
trivy k8s -n security-insights  deploy/name-of-my-deployment --scanners vuln --format json > out.json
```

DefectDojo can parse the findings from both containers in the deployment and reports on some Python package in the in ofthe containers 👍

**Comment by J12934** [comment]
@nixboot ok thats really good, especially as our generic mapping for the trivy-k8s scan apparently is broken right now :D
Changing it to use the DefectDojo mapping should then also resolve #2324

**Comment by nixboot** [comment]
Oh. Yikes 😅 Is this something you have time to look into soon, or should we dust off our Java skills?

**Comment by Weltraumschaf** [comment]
@nixboot We're working on it. The fix itself is very simple. We're discussing at the moment what a good URL would be for this scanner.

**Comment by nixboot** [comment]
@Weltraumschaf Thanks for the update. I have an update aswell - albeit it being a bit late.

I took the hints @J12934 mentioned, and tried changing the mappings in `ScanNameMapping` in https://github.com/secureCodeBox/secureCodeBox/blob/main/hooks/persistence-defectdojo/hook/src/main/java/io/securecodebox/persistence/util/ScanNameMapping.java#L17. I've added the following

```java
  TRIVY_IMAGE("trivy-image", ScanType.TRIVY_SCAN),
  TRIVY_K8S("trivy-k8s", ScanType.TRIVY_SCAN),
  TRIVY_FILESYSTEM("trivy-filesystem", ScanType.TRIVY_SCAN),
  TRIVY_REPO("trivy-repo", ScanType.TRIVY_SCAN),
```

And that works 🚀 ... At least on the SecureCodeBox side.

Unfortunately, the Trivy scans I have produced using the SCB ScanType `trivy-k8s` do not fill in the `ClusterName` in the output. It is just empty

```json
{
  "ClusterName": "",
  "Resources": [
    {
      "Namespace": "test-namespace",
      "Kind": "Deployment",
      "Name": "my-deployment-name",
...
```

This has the unfortunate effect on the Trivy parser in DefectDojo that it reports "Schema of Trivy json report is not supported".

The parser has a check on `ClusterName`, but interprets an empty string as the same as the object value `ClusterName` not being there at all (see the ifcase in the parser here https://github.com/DefectDojo/django-DefectDojo/blob/master/dojo/tools/trivy/parser.py#L94).

If I, behind the back of DefectDojo, modified the parser, and replaces
```python
elif cluster_name:
```

with 

```python
elif cluster_name is not None:
```

the scanning result is successfully uploaded.
